### PR TITLE
Add support for boost 1.66

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

**Comment by jschueller** [comment]
@conda-forge-admin, please rerender

File: recipe/bld.bat [hunk 1]
```diff
@@ -17,3 +17,4 @@ move share %LIBRARY_PREFIX%\
 if errorlevel 1 exit 1
 
 cp %RECIPE_DIR%\Windows-Flang-Fortran.cmake %LIBRARY_PREFIX%\share\cmake-3.10\Modules\Platform\
+cp %RECIPE_DIR%\FindBoost.cmake %LIBRARY_PREFIX%\share\cmake-3.10\Modules\
```
Comment (isuruf) [inline recipe/bld.bat]: Is both patching and copying necessary?

Comment (jschueller) [reply recipe/bld.bat]: We could copy or patch for both source and binary releases. I started with the patch the realized the windows package is binary-only so I just copied the patched module for windows.

Comment (isuruf) [reply recipe/bld.bat]: Ah, okay

**Pull request merged**